lib/ReplaceOpenCLBuiltinPass.cpp

```diff
@@ -2682,14 +2682,26 @@ bool ReplaceOpenCLBuiltinPass::replaceVstoreHalf4(Module &M) {
   });
 }
 
-bool ReplaceOpenCLBuiltinPass::replaceReadImageF(Module &M) {
+bool ReplaceOpenCLBuiltinPass::replaceSampledReadImageWithIntCoords(Module &M) {
   bool Changed = false;
 
   const std::map<const char *, const char *> Map = {
+      // TODO 1D, 1Darray
+      // 2D
+      {"_Z11read_imagei14ocl_image2d_ro11ocl_samplerDv2_i",
+       "_Z11read_imagei14ocl_image2d_ro11ocl_samplerDv2_f"},
+      {"_Z12read_imageui14ocl_image2d_ro11ocl_samplerDv2_i",
+       "_Z12read_imageui14ocl_image2d_ro11ocl_samplerDv2_f"},
       {"_Z11read_imagef14ocl_image2d_ro11ocl_samplerDv2_i",
        "_Z11read_imagef14ocl_image2d_ro11ocl_samplerDv2_f"},
-      {"_Z11read_imagef14ocl_image2d_ro11ocl_samplerDv4_i",
```

Comment:
This should correspond to a 2D array image read. It's ok to leave this as a todo.

Reply:
Apparently not. A quick test suggests that Clang mangles these to `ocl_image2d_array_ro` which matches my reading of https://github.com/llvm/llvm-project/blob/master/clang/lib/AST/ItaniumMangle.cpp#L2682. We'll find out soon enough if it did correspond to an existing built-in :).